src/bitarray.ts

```diff
@@ -172,6 +172,86 @@ export default class BitArray extends BitTypedArray {
         return ret;
     }
 
+    /**
+     *
+     * @param charSet a set of n characters to use to encode the BitArray; charSet.length must be a power of 2 (2, 4, 8, etc)
+     * The more characters in the set, the more compact the resulting output will be
+     * @returns a string encoded using the provided character set (e.g., base64 encoding can be achieved with this)
+     */
+    encodeWithCharacterSet( charSet: string | string[] ): string {
```

Comment:
`string[]` would work, but is it needed at all? I would suggest remove and keep `string` only.

Reply:
Hmm, it does remove an avenue of optimization if the list is already stored as a char[], but since js doesn't have that type, I agree.